doc/user/content/ops/optimization.md

```diff
@@ -79,18 +90,86 @@ materialize=> EXPLAIN SELECT * FROM t1, t2, t3 WHERE t1.y = t2.x AND t2.y = t3.x
                                                  +
  Used Indexes:                                   +
    - materialize.public.t1_y                     +
-   - materialize.public.t3_x                     +
    - materialize.public.t2_x                     +
-   - materialize.public.t2_y                     +
+   - materialize.public.t2_w                     +
+   - materialize.public.t3_z                     +
 ```
-Delta joins have the advantage of using negligible additional memory outside the explicitly created indexes on the inputs. For more details, see [Maintaining Joins using Few Resources](https://materialize.com/blog/maintaining-joins-using-few-resources).
+Delta joins have the advantage of using negligible additional memory outside the explicitly created indexes on the inputs. For more details, see [Delta Joins and Late Materialization](/overview/delta-joins).
 
 If your query filters one or more of the join inputs by a literal equality (e.g., `t1.y = 42`), place one of those inputs first in the `FROM` clause. In particular, this can speed up [ad hoc `SELECT` queries](/sql/select/#ad-hoc-queries) by accessing inputs using index lookups, rather than full scans.
 
-### Default
-
-Create a default index when there is no particular `WHERE` or `JOIN` clause that would fit the above cases. This can still speed up your query by reading the input from memory.
+#### Further Optimize with Late Materialization
+
+Even with delta joins, Materialize can potentially maintain much more data in memory than is needed. You can further optimize your multi-way joins with late materialization using this recipe:
```

Comment:
```suggestion
Materialize can further optimize memory usage - using a pattern known as late materialization - when joining relations containing primary and foreign key constraints.

```

We should try and phrase things positiviely. DeltaQueryies don't maintain much more data in memory than is needed but instead we can futher optimize certain patterns 😄